core/src/main/scala-3/cats/instances/iarray.scala

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (c) 2015 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package cats
+package instances
+
+import cats.data.{Chain, Ior}
+import cats.instances.StaticMethods.{appendAll, appendAllMixed}
+import cats.kernel.compat.scalaVersionSpecific.*
+
+import scala.annotation.tailrec
+import scala.reflect.ClassTag
+import scala.collection.mutable.Builder
+
+trait IArrayInstances extends cats.kernel.instances.IArrayInstances {
+  implicit val catsStdInstancesForIArray
+    : Traverse[IArray] & Monad[IArray] & Alternative[IArray] & CoflatMap[IArray] & Align[IArray] =
+    new Traverse[IArray] with Monad[IArray] with Alternative[IArray] with CoflatMap[IArray] with Align[IArray] {
+
+      def empty[A]: IArray[A] = {
+        implicit val fakeClassTag: ClassTag[A] = summonFakeTag[A]
+        IArray.empty[A]
```

Comment:
`IArray` is defined as covariant:
https://github.com/scala/scala3/blob/b8afeea3de732d0cfccaa372e38af1bc02f7eb31/library/src/scala/IArray.scala#L8

So, I think we can return `empty[Nothing]` here and it will type check without any casts.